src/bench/gcs_filter.cpp

```diff
@@ -32,12 +32,65 @@ static void MatchGCSFilter(benchmark::Bench& bench)
         element[1] = static_cast<unsigned char>(i >> 8);
         elements.insert(std::move(element));
     }
-    GCSFilter filter({0, 0, 20, 1 << 20}, elements);
+    GCSFilter filter({0, 0, BASIC_FILTER_P, BASIC_FILTER_M}, elements);
 
     bench.unit("elem").run([&] {
         filter.Match(GCSFilter::Element());
     });
 }
 
+static void DecodeGCSFilter(benchmark::Bench& bench)
+{
+    GCSFilter::ElementSet elements;
+    for (int i = 0; i < 10000; ++i) {
+        GCSFilter::Element element(32);
+        element[0] = static_cast<unsigned char>(i);
+        element[1] = static_cast<unsigned char>(i >> 8);
+        elements.insert(std::move(element));
+    }
+    GCSFilter filter({0, 0, BASIC_FILTER_P, BASIC_FILTER_M}, elements);
+    auto encoded = filter.GetEncoded();
+
+    bench.unit("elem").run([&] {
+        GCSFilter filter({0, 0, BASIC_FILTER_P, BASIC_FILTER_M}, encoded);
+    });
+}
+
+static void BlockFilterGetHash(benchmark::Bench& bench)
+{
+    GCSFilter::ElementSet elements;
+    for (int i = 0; i < 10000; ++i) {
+        GCSFilter::Element element(32);
+        element[0] = static_cast<unsigned char>(i);
+        element[1] = static_cast<unsigned char>(i >> 8);
+        elements.insert(std::move(element));
```

Comment:
It looks like there is a fair amount of duplicate boilerplate in each of these benchmarks that you could extract out to a common setup helper in a separate commit, if you like. See  `EvictionProtectionCommon()` in `src/bench/peer_eviction.cpp` for an example.

Reply:
Since most of the duplicate code is just generating  the elements, how's separating that out to a function like so:

```
static const GCSFilter::ElementSet GenerateGCSTestElements(){
    GCSFilter::ElementSet elements;
    for (int i = 0; i < 10000; ++i) {
        GCSFilter::Element element(32);
        element[0] = static_cast<unsigned char>(i);
        element[1] = static_cast<unsigned char>(i >> 8);
        elements.insert(std::move(element));
    }

    return elements;
}
```

And having each of the benchmarks call that function?

Reply:
I've applied the above change I suggested above.